### PR TITLE
Feature/obg 901 hbci wrong pin handling

### PR DESCRIPTION
**Pull request by electronicpeter**

### CODE REVIEW

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java [hunk 1]
```diff
@@ -521,11 +497,14 @@ public SELF user_selected_sca_challenge_type_email1_to_embedded_authorization()
         return self();
     }
 
-    public SELF user_max_musterman_provided_correct_sca_challenge_result_after_wrong_to_embedded_authorization_and_sees_redirect_to_fintech_ok() {
-        assertThat(this.redirectUriToGetUserParams).contains("sca-result").contains("/EMAIL").contains("wrong=true");
+    public void user_max_musterman_provided_correct_sca_challenge_result_after_wrong_to_embedded_authorization_and_sees_redirect_to_fintech_ok() {
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:500]: Please return `SELF` as it makes this step `continuable` within `AccountInformationRequestCommon` class

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java [hunk 2]
```diff
@@ -539,11 +518,10 @@ public SELF user_max_musterman_provided_sca_challenge_result_to_embedded_authori
         return user_max_musterman_provided_sca_challenge_result_to_embedded_authorization_and_sees_redirect_to_fintech_ok("/EMAIL");
     }
 
-    public SELF user_provided_sca_challenge_result_to_embedded_authorization_and_sees_redirect_to_fintech_ok() {
+    public void user_provided_sca_challenge_result_to_embedded_authorization_and_sees_redirect_to_fintech_ok() {
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:521]: Please return `SELF` as it makes this step `continuable` within `AccountInformationRequestCommon` class

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciAuthorizationPossibleErrorHandler.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        }
+        if (ex.getMultibankingError().equals(MultibankingError.INVALID_TAN)) {
+            log.warn("wrong tan was entered");
+            return;
+        }
+        if (ex.getMultibankingError().equals(MultibankingError.HBCI_ERROR)) {
+            log.warn("hbci error");
+            return;
+        }
+        log.error("dont know this error: {}", ex.getMultibankingError().name());
```
Comment (valb3r) [inline opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciAuthorizationPossibleErrorHandler.java:51]: Please use `Unknown error, can't handle: {}`

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciInitiateSendPinAndPsuId.java [hunk 4]
```diff
@@ -20,39 +22,62 @@
...
-            );
-        }
 
-        UpdatePsuAuthenticationRequest request = new UpdatePsuAuthenticationRequest();
-        request.setCredentials(consent.getCredentials());
-        request.setBankApiConsentData(consent);
-        request.setBank(context.getBank());
+        errorSink.handlePossibleAuthorizationError(
+                () -> {
+                    context.setWrongAuthCredentials(false);
```
Comment (valb3r) [inline opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciInitiateSendPinAndPsuId.java:38]: Suggest to extract (Runnable) lambda function body to the private method as it is quite long

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciSendTanChallenge.java [hunk 5]
```diff
@@ -20,21 +21,43 @@
...
 
     @Override
     protected void doRealExecution(DelegateExecution execution, HbciContext context) {
-        HbciConsent consent = context.getHbciDialogConsent();
 
-        TransactionAuthorisationRequest request = create(new BankApiUser(), new BankAccess(), context.getBank(), consent);
-        request.setScaAuthenticationData(context.getPsuTan());
+        errorSink.handlePossibleAuthorizationError(
+                () -> {
+                    context.setWrongAuthCredentials(false);
```
Comment (valb3r) [inline opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciSendTanChallenge.java:35]: Suggest to extract (Runnable) lambda function body to the private method as it is quite long
It is a smell as well as causes problems handling non-runtime exceptions:
https://developer.ibm.com/articles/j-java8idioms6/

**Review by valb3r — COMMENTED** [review]
Ok, minor issues

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java [hunk 6]
```diff
@@ -265,7 +254,7 @@ public SELF user_anton_brueckner_provided_initial_parameters_to_list_accounts_wi
         return self();
     }
 
-    public SELF user_denied_consent() {
+    public void user_denied_consent() {
```
[the remark this answers is not in the document]
Comment (electronicpeter) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java]: done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java (hunk 1 above)
Comment (electronicpeter) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java]: done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java (hunk 2 above)
Comment (electronicpeter) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:523]: done

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciAuthorizationPossibleErrorHandler.java (hunk 3 above)
Comment (electronicpeter) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciAuthorizationPossibleErrorHandler.java]: done

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciInitiateSendPinAndPsuId.java (hunk 4 above)
Comment (electronicpeter) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciInitiateSendPinAndPsuId.java]: done

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciSendTanChallenge.java (hunk 5 above)
Comment (electronicpeter) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciSendTanChallenge.java]: done. Thanks for link!

**Review by valb3r — APPROVED** [review]

**Pull request merged**